core/src/com/unciv/logic/GameInfo.kt

```diff
@@ -499,7 +499,12 @@ class GameInfo : IsPartOfGameInfoSerialization, HasGameInfoSerializationVersion
 
         convertFortify()
 
-        for (civInfo in civilizations) {
+        sequence {
+            // update city-state resource first since the happiness of major civ depends on it.
+            // See issue: https://github.com/yairm210/Unciv/issues/7781
+            yieldAll(civilizations.filter { it.isCityState() })
+            yieldAll(civilizations.filter { !it.isCityState(); it.isMajorCiv() })
```

Comment:
By definition any civ that is a major civ cannot be a city state, so you can remove the first function call here